controllers/gateway_controller.go

```diff
@@ -149,7 +150,7 @@ func (r *GatewayReconciler) reconcile(ctx context.Context, req ctrl.Request) err
 					continue
 				}
 				gwName := types.NamespacedName{
-					Namespace: httpRoute.Namespace,
+					Namespace: gw.Namespace,
```

Comment:
this needs to be `httpRoute.Spec.ParentRefs[0].Namespace`  if exist, or default to httpRoute.Namespace

Reply:
We need a e2e to cover this